File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        window_sum = 0
        max_average = float('-inf')
        
        for i in range(k):
            window_sum += nums[i]
        max_average = window_sum / k
        
        for i in range(k, len(nums)):
            window_sum += nums[i] - nums[i - k]
            current_average = window_sum / k
            max_average = max(max_average, current_average)
        
        return max_average
            
        

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
    let left = 0
    let currSum = 0
    let answer = 0
    for(let right = 0; right < k ; right++){
        currSum+=nums[right]
    }
    answer = currSum/k
    for(let i = k;i < nums.length; i++){
        currSum+=nums[i]
        currSum-=nums[left]
        left++
        answer = Math.max(answer,currSum/k)
    }
    return answer
    
};